### PR TITLE
[ci/release] GCE variants for Alpa, Golden notebooks, Lightning, Horovod, Workspace templates

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by can-anyscale — APPROVED** [review]
gogogo

File: release/ray_release/config.py [hunk 1]
```diff
@@ -135,6 +135,13 @@ def validate_release_test_collection(
             )
             num_errors += 1
 
+        error = validate_test_cluster_env(test)
```
Comment (can-anyscale) [inline release/ray_release/config.py:138]: nice!

**Review by can-anyscale — APPROVED** [review]

**Comment by krfricke** [comment]
So the code is now working.

But the test is failing due to timeouts. I'll open an issue to track.

**Review by maxpumperla — APPROVED** [review]
approving docs changes

**Pull request merged**